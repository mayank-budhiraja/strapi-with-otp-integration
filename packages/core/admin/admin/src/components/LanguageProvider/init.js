import localStorageKey from './utils/localStorageKey';

const init = localeNames => {
  const languageFromLocaleStorage = window.localStorage.getItem(localStorageKey);
  const appLanguage = localeNames[languageFromLocaleStorage] ? languageFromLocaleStorage : 'en';

  return {
    locale: appLanguage,
    localeNames,
  };
};

export default init;
