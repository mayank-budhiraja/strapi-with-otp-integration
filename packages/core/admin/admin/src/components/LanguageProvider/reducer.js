/*
 *
 * LanguageProvider reducer
 *
 */

const initialState = {
  localeNames: { en: 'English' },
  locale: 'en',
};

const languageProviderReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'CHANGE_LOCALE': {
      const { locale } = action;

      if (!state.localeNames[locale]) {
        return state;
      }

      return { ...state, locale };
    }
    default: {
      return state;
    }
  }
};

export default languageProviderReducer;
export { initialState };
