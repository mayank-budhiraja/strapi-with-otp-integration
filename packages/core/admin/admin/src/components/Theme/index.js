import React from 'react';
import { ThemeProvider } from '@strapi/design-system/ThemeProvider';
import PropTypes from 'prop-types';
import { useThemeToggle } from '../../hooks';
import GlobalStyle from '../GlobalStyle';

const Theme = ({ children }) => {
  const { currentTheme, themes } = useThemeToggle();

  return (
    <ThemeProvider theme={themes[currentTheme] || themes.light}>
      {children}
      <GlobalStyle />
    </ThemeProvider>
  );
};

Theme.propTypes = {
  children: PropTypes.element.isRequired,
};

export default Theme;
