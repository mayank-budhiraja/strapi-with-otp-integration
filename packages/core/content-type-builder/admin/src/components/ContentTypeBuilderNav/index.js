import React from 'react';
import upperFirst from 'lodash/upperFirst';
import { useIntl } from 'react-intl';
import { NavLink } from 'react-router-dom';
import Plus from '@strapi/icons/Plus';
import { Box } from '@strapi/design-system/Box';
import {
  SubNav,
  SubNavHeader,
  SubNavLink,
  SubNavLinkSection,
  SubNavSection,
  SubNavSections,
} from '@strapi/design-system/v2/SubNav';
import { TextButton } from '@strapi/design-system/TextButton';
import useContentTypeBuilderMenu from './useContentTypeBuilderMenu';
import getTrad from '../../utils/getTrad';

const ContentTypeBuilderNav = () => {
  const { menu, searchValue, onSearchChange } = useContentTypeBuilderMenu();
  const { formatMessage } = useIntl();

  return (
    <SubNav
      ariaLabel={formatMessage({
        id: `${getTrad('plugin.name')}`,
        defaultMessage: 'Content-Types Builder',
      })}
    >
      <SubNavHeader
        searchable
        value={searchValue}
        onClear={() => onSearchChange('')}
        onChange={e => onSearchChange(e.target.value)}
        label={formatMessage({
          id: `${getTrad('plugin.name')}`,
          defaultMessage: 'Content-Types Builder',
        })}
        searchLabel="Search..."
      />
      <SubNavSections>
        {menu.map(section => {
          const title = section.title.id;

          return (
            <React.Fragment key={section.name}>
              <SubNavSection
                label={formatMessage({ id: title, defaultMessage: title })}
                collapsable
                badgeLabel={section.links.length.toString()}
              >
                {section.links.map(link => {
                  if (link.links) {
                    return (
                      <SubNavLinkSection key={link.name} label={upperFirst(link.title)}>
                        {link.links.map(subLink => (
                          <SubNavLink
                            as={NavLink}
                            to={subLink.to}
                            active={subLink.active}
                            key={subLink.name}
                          >
                            {upperFirst(
                              formatMessage({ id: subLink.name, defaultMessage: subLink.title })
                            )}
                          </SubNavLink>
                        ))}
                      </SubNavLinkSection>
                    );
                  }

                  return (
                    <SubNavLink as={NavLink} to={link.to} active={link.active} key={link.name}>
                      {upperFirst(formatMessage({ id: link.name, defaultMessage: link.title }))}
                    </SubNavLink>
                  );
                })}
              </SubNavSection>
              {section.customLink && (
                <Box paddingLeft={7}>
                  <TextButton onClick={section.customLink.onClick} startIcon={<Plus />}>
                    {formatMessage({
                      id: section.customLink.id,
                      defaultMessage: section.customLink.defaultMessage,
                    })}
                  </TextButton>
                </Box>
              )}
            </React.Fragment>
          );
        })}
      </SubNavSections>
    </SubNav>
  );
};

export default ContentTypeBuilderNav;
