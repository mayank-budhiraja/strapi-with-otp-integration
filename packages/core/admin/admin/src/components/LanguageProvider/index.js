/*
 *
 * LanguageProvider
 *
 * this component connects the redux state language locale to the
 * IntlProvider component and i18n messages (loaded from `app/translations`)
 */

import React, { useEffect, useReducer } from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';
import defaultsDeep from 'lodash/defaultsDeep';
import LocalesProvider from '../LocalesProvider';
import localStorageKey from './utils/localStorageKey';
import init from './init';
import reducer, { initialState } from './reducer';

const LanguageProvider = ({ children, localeNames, messages }) => {
  const [{ locale }, dispatch] = useReducer(reducer, initialState, () => init(localeNames));

  useEffect(() => {
    // Set user language in local storage.
    window.localStorage.setItem(localStorageKey, locale);
  }, [locale]);

  const changeLocale = locale => {
    dispatch({
      type: 'CHANGE_LOCALE',
      locale,
    });
  };

  const appMessages = defaultsDeep(messages[locale], messages.en);

  return (
    <IntlProvider locale={locale} defaultLocale="en" messages={appMessages} textComponent="span">
      <LocalesProvider changeLocale={changeLocale} localeNames={localeNames} messages={appMessages}>
        {children}
      </LocalesProvider>
    </IntlProvider>
  );
};

LanguageProvider.propTypes = {
  children: PropTypes.element.isRequired,
  localeNames: PropTypes.objectOf(PropTypes.string).isRequired,
  messages: PropTypes.object.isRequired,
};

export default LanguageProvider;
