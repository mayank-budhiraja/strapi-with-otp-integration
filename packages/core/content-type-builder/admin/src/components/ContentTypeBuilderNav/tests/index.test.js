/**
 *
 * Tests for ContentTypeBuilderNav
 *
 */

import { Layout, lightTheme, darkTheme } from '@strapi/design-system';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import LanguageProvider from '../../../../../../admin/admin/src/components/LanguageProvider';
import Theme from '../../../../../../admin/admin/src/components/Theme';
import ThemeToggleProvider from '../../../../../../admin/admin/src/components/ThemeToggleProvider';
import en from '../../../../../../admin/admin/src/translations/en.json';
import ContentTypeBuilderNav from '../index';
import mockData from './mockData';

jest.mock('../useContentTypeBuilderMenu.js', () => {
  return jest.fn(() => ({
    menu: mockData,
    searchValue: '',
    onSearchChange: () => {},
  }));
});

const makeApp = () => {
  const history = createMemoryHistory();
  const messages = { en };
  const localeNames = { en: 'English' };

  return (
    <LanguageProvider messages={messages} localeNames={localeNames}>
      <ThemeToggleProvider themes={{ light: lightTheme, dark: darkTheme }}>
        <Theme>
          <Router history={history}>
            <Layout sideNav={<ContentTypeBuilderNav />}>
              <div />
            </Layout>
          </Router>
        </Theme>
      </ThemeToggleProvider>
    </LanguageProvider>
  );
};

describe('<ContentTypeBuilderNav />', () => {
  it('renders and matches the snapshot', () => {
    const App = makeApp(ContentTypeBuilderNav);
    const { container } = render(App);

    expect(container).toMatchSnapshot();
  });
});
