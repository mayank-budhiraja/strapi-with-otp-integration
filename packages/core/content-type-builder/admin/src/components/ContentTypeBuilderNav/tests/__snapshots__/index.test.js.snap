// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentTypeBuilderNav /> renders and matches the snapshot 1`] = `
.c41 {
  padding-bottom: 56px;
}

.c0 {
  display: grid;
  grid-template-columns: auto 1fr;
}

.c42 {
  overflow-x: hidden;
}

.c43 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c31 {
  padding-left: 32px;
}

.c2 {
  padding-top: 24px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-left: 24px;
}

.c8 {
  padding-top: 16px;
}

.c9 {
  background: #eaeaef;
}

.c12 {
  padding-top: 8px;
  padding-bottom: 16px;
}

.c15 {
  padding-top: 8px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-left: 24px;
}

.c20 {
  padding-right: 4px;
}

.c23 {
  background: #eaeaef;
  padding: 4px;
  border-radius: 4px;
  min-width: 20px;
}

.c26 {
  background: #f6f6f9;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 32px;
}

.c29 {
  padding-left: 8px;
}

.c36 {
  padding-top: 8px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-left: 32px;
}

.c1 {
  width: 14.5rem;
  background: #f6f6f9;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  height: 100vh;
  overflow-y: auto;
  border-right: 1px solid #dcdce4;
  z-index: 1;
}

.c3 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c13 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c17 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c18 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c24 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  color: #32324d;
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.22;
}

.c21 {
  color: #666687;
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
}

.c30 {
  color: #32324d;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c40 {
  font-weight: 500;
  color: #32324d;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c6 svg {
  height: 12px;
  width: 12px;
}

.c6 svg > g,
.c6 svg path {
  fill: #ffffff;
}

.c6[aria-disabled='true'] {
  pointer-events: none;
}

.c6:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c6:focus-visible {
  outline: none;
}

.c6:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
}

.c7 svg > g,
.c7 svg path {
  fill: #8e8ea9;
}

.c7:hover svg > g,
.c7:hover svg path {
  fill: #666687;
}

.c7:active svg > g,
.c7:active svg path {
  fill: #a5a5ba;
}

.c7[aria-disabled='true'] {
  background-color: #eaeaef;
}

.c7[aria-disabled='true'] svg path {
  fill: #666687;
}

.c10 {
  height: 1px;
  border: none;
  margin: 0;
}

.c11 {
  width: 1.5rem;
  background-color: #dcdce4;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #32324d;
}

.c27 svg > * {
  fill: #666687;
}

.c27.active {
  background-color: #f0f0ff;
  border-right: 2px solid #4945ff;
}

.c27.active svg > * {
  fill: #271fe0;
}

.c27.active .c4 {
  color: #271fe0;
  font-weight: 500;
}

.c27:focus-visible {
  outline-offset: -2px;
}

.c28 {
  width: 0.75rem;
  height: 0.25rem;
}

.c28 * {
  fill: #666687;
}

.c37 {
  max-height: 2rem;
}

.c37 svg {
  height: 0.25rem;
}

.c37 svg path {
  fill: #4a4a6a;
}

.c38 {
  border: none;
  padding: 0;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c39 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 0.75rem;
  -webkit-transform: rotateX(0deg);
  -ms-transform: rotateX(0deg);
  transform: rotateX(0deg);
}

.c19 {
  border: none;
  padding: 0;
  background: transparent;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: rotateX(0deg);
  -ms-transform: rotateX(0deg);
  transform: rotateX(0deg);
}

.c16 {
  max-height: 2rem;
}

.c16 svg {
  height: 0.25rem;
}

.c16 svg path {
  fill: #8e8ea9;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c14 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c14 > * + * {
  margin-top: 8px;
}

.c34 {
  padding-right: 8px;
}

.c35 {
  color: #4945ff;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c32 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c33 {
  background: transparent;
  border: none;
  position: relative;
  outline: none;
}

.c33[aria-disabled='true'] {
  pointer-events: none;
}

.c33[aria-disabled='true'] svg path {
  fill: #666687;
}

.c33 svg {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.625rem;
}

.c33 svg path {
  fill: #4945ff;
}

.c33:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c33:focus-visible {
  outline: none;
}

.c33:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

<div>
  <div
    class="c0"
  >
    <nav
      aria-label="Content-Types Builder"
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Content-Types Builder
          </h2>
          <span>
            <button
              aria-disabled="false"
              aria-labelledby="tooltip-2"
              class="c6 c7"
              tabindex="0"
              type="button"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M23.813 20.163l-5.3-5.367a9.792 9.792 0 001.312-4.867C19.825 4.455 15.375 0 9.913 0 4.45 0 0 4.455 0 9.929c0 5.473 4.45 9.928 9.912 9.928a9.757 9.757 0 005.007-1.4l5.275 5.35a.634.634 0 00.913 0l2.706-2.737a.641.641 0 000-.907zM9.91 3.867c3.338 0 6.05 2.718 6.05 6.061s-2.712 6.061-6.05 6.061c-3.337 0-6.05-2.718-6.05-6.06 0-3.344 2.713-6.062 6.05-6.062z"
                  fill="#32324D"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </span>
        </div>
        <div
          class="c8"
        >
          <hr
            class="c9 c10 c11"
          />
        </div>
      </div>
      <div
        class="c12"
      >
        <ul
          class="c13 c14"
          spacing="2"
        >
          <li>
            <div
              class=""
            >
              <div
                class="c15 c16"
              >
                <div
                  class="c17"
                >
                  <button
                    aria-controls="subnav-list-4"
                    aria-expanded="true"
                    class="c18 c19"
                  >
                    <div
                      class="c20"
                    >
                      <span
                        class="c4 c21"
                      >
                        content-type-builder.menu.section.models.name.
                      </span>
                    </div>
                    <div
                      class="c22"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        height="1em"
                        viewBox="0 0 14 8"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                          fill="#32324D"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </button>
                  <div
                    class="c23 c24 c25"
                  >
                    <span
                      class="c4 c21"
                    >
                      2
                    </span>
                  </div>
                </div>
              </div>
              <ul
                id="subnav-list-4"
              >
                <li>
                  <a
                    class="c26 c27"
                    href="/plugins/content-type-builder/content-types/application::address.address"
                  >
                    <div
                      class="c18"
                    >
                      <svg
                        class="c28"
                        fill="none"
                        height="1em"
                        viewBox="0 0 4 4"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="#A5A5BA"
                          height="4"
                          rx="2"
                          width="4"
                        />
                      </svg>
                      <div
                        class="c29"
                      >
                        <span
                          class="c4 c30"
                        >
                          Address
                        </span>
                      </div>
                    </div>
                  </a>
                </li>
                <li>
                  <a
                    class="c26 c27"
                    href="/plugins/content-type-builder/content-types/application::category.category"
                  >
                    <div
                      class="c18"
                    >
                      <svg
                        class="c28"
                        fill="none"
                        height="1em"
                        viewBox="0 0 4 4"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="#A5A5BA"
                          height="4"
                          rx="2"
                          width="4"
                        />
                      </svg>
                      <div
                        class="c29"
                      >
                        <span
                          class="c4 c30"
                        >
                          Category
                        </span>
                      </div>
                    </div>
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="c31"
            >
              <button
                aria-disabled="false"
                class="c32 c33"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="c34"
                >
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M24 13.604a.3.3 0 01-.3.3h-9.795V23.7a.3.3 0 01-.3.3h-3.21a.3.3 0 01-.3-.3v-9.795H.3a.3.3 0 01-.3-.3v-3.21a.3.3 0 01.3-.3h9.795V.3a.3.3 0 01.3-.3h3.21a.3.3 0 01.3.3v9.795H23.7a.3.3 0 01.3.3v3.21z"
                      fill="#212134"
                    />
                  </svg>
                </span>
                <span
                  class="c35"
                >
                  Create new collection type
                </span>
              </button>
            </div>
          </li>
          <li>
            <div
              class=""
            >
              <div
                class="c15 c16"
              >
                <div
                  class="c17"
                >
                  <button
                    aria-controls="subnav-list-5"
                    aria-expanded="true"
                    class="c18 c19"
                  >
                    <div
                      class="c20"
                    >
                      <span
                        class="c4 c21"
                      >
                        content-type-builder.menu.section.single-types.name.
                      </span>
                    </div>
                    <div
                      class="c22"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        height="1em"
                        viewBox="0 0 14 8"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                          fill="#32324D"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </button>
                  <div
                    class="c23 c24 c25"
                  >
                    <span
                      class="c4 c21"
                    >
                      1
                    </span>
                  </div>
                </div>
              </div>
              <ul
                id="subnav-list-5"
              >
                <li>
                  <a
                    class="c26 c27"
                    href="/plugins/content-type-builder/content-types/application::homepage.homepage"
                  >
                    <div
                      class="c18"
                    >
                      <svg
                        class="c28"
                        fill="none"
                        height="1em"
                        viewBox="0 0 4 4"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          fill="#A5A5BA"
                          height="4"
                          rx="2"
                          width="4"
                        />
                      </svg>
                      <div
                        class="c29"
                      >
                        <span
                          class="c4 c30"
                        >
                          Homepage
                        </span>
                      </div>
                    </div>
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="c31"
            >
              <button
                aria-disabled="false"
                class="c32 c33"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="c34"
                >
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M24 13.604a.3.3 0 01-.3.3h-9.795V23.7a.3.3 0 01-.3.3h-3.21a.3.3 0 01-.3-.3v-9.795H.3a.3.3 0 01-.3-.3v-3.21a.3.3 0 01.3-.3h9.795V.3a.3.3 0 01.3-.3h3.21a.3.3 0 01.3.3v9.795H23.7a.3.3 0 01.3.3v3.21z"
                      fill="#212134"
                    />
                  </svg>
                </span>
                <span
                  class="c35"
                >
                  Create new single type
                </span>
              </button>
            </div>
          </li>
          <li>
            <div
              class=""
            >
              <div
                class="c15 c16"
              >
                <div
                  class="c17"
                >
                  <button
                    aria-controls="subnav-list-6"
                    aria-expanded="true"
                    class="c18 c19"
                  >
                    <div
                      class="c20"
                    >
                      <span
                        class="c4 c21"
                      >
                        content-type-builder.menu.section.components.name.
                      </span>
                    </div>
                    <div
                      class="c22"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        height="1em"
                        viewBox="0 0 14 8"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                          fill="#32324D"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </div>
                  </button>
                  <div
                    class="c23 c24 c25"
                  >
                    <span
                      class="c4 c21"
                    >
                      2
                    </span>
                  </div>
                </div>
              </div>
              <ul
                id="subnav-list-6"
              >
                <li>
                  <div
                    class=""
                  >
                    <div
                      class="c36 c37"
                    >
                      <div
                        class="c17"
                      >
                        <button
                          aria-controls="subnav-list-7"
                          aria-expanded="true"
                          class="c38"
                        >
                          <div
                            class="c39"
                          >
                            <svg
                              aria-hidden="true"
                              fill="none"
                              height="1em"
                              viewBox="0 0 14 8"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                                fill="#32324D"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </div>
                          <div
                            class="c29"
                          >
                            <span
                              class="c4 c40"
                            >
                              Basic
                            </span>
                          </div>
                        </button>
                      </div>
                    </div>
                    <ul
                      id="subnav-list-7"
                    >
                      <li>
                        <a
                          class="c26 c27"
                          href="/plugins/content-type-builder/component-categories/basic/basic.simple"
                        >
                          <div
                            class="c18"
                          >
                            <svg
                              class="c28"
                              fill="none"
                              height="1em"
                              viewBox="0 0 4 4"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <rect
                                fill="#A5A5BA"
                                height="4"
                                rx="2"
                                width="4"
                              />
                            </svg>
                            <div
                              class="c29"
                            >
                              <span
                                class="c4 c30"
                              >
                                Simple
                              </span>
                            </div>
                          </div>
                        </a>
                      </li>
                    </ul>
                  </div>
                </li>
                <li>
                  <div
                    class=""
                  >
                    <div
                      class="c36 c37"
                    >
                      <div
                        class="c17"
                      >
                        <button
                          aria-controls="subnav-list-8"
                          aria-expanded="true"
                          class="c38"
                        >
                          <div
                            class="c39"
                          >
                            <svg
                              aria-hidden="true"
                              fill="none"
                              height="1em"
                              viewBox="0 0 14 8"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                                fill="#32324D"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </div>
                          <div
                            class="c29"
                          >
                            <span
                              class="c4 c40"
                            >
                              Default
                            </span>
                          </div>
                        </button>
                      </div>
                    </div>
                    <ul
                      id="subnav-list-8"
                    >
                      <li>
                        <a
                          class="c26 c27"
                          href="/plugins/content-type-builder/component-categories/default/default.closingperiod"
                        >
                          <div
                            class="c18"
                          >
                            <svg
                              class="c28"
                              fill="none"
                              height="1em"
                              viewBox="0 0 4 4"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <rect
                                fill="#A5A5BA"
                                height="4"
                                rx="2"
                                width="4"
                              />
                            </svg>
                            <div
                              class="c29"
                            >
                              <span
                                class="c4 c30"
                              >
                                Closingperiod
                              </span>
                            </div>
                          </div>
                        </a>
                      </li>
                      <li>
                        <a
                          class="c26 c27"
                          href="/plugins/content-type-builder/component-categories/default/default.dish"
                        >
                          <div
                            class="c18"
                          >
                            <svg
                              class="c28"
                              fill="none"
                              height="1em"
                              viewBox="0 0 4 4"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <rect
                                fill="#A5A5BA"
                                height="4"
                                rx="2"
                                width="4"
                              />
                            </svg>
                            <div
                              class="c29"
                            >
                              <span
                                class="c4 c30"
                              >
                                Dish
                              </span>
                            </div>
                          </div>
                        </a>
                      </li>
                    </ul>
                  </div>
                </li>
              </ul>
            </div>
            <div
              class="c31"
            >
              <button
                aria-disabled="false"
                class="c32 c33"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="c34"
                >
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M24 13.604a.3.3 0 01-.3.3h-9.795V23.7a.3.3 0 01-.3.3h-3.21a.3.3 0 01-.3-.3v-9.795H.3a.3.3 0 01-.3-.3v-3.21a.3.3 0 01.3-.3h9.795V.3a.3.3 0 01.3-.3h3.21a.3.3 0 01.3.3v9.795H23.7a.3.3 0 01.3.3v3.21z"
                      fill="#212134"
                    />
                  </svg>
                </span>
                <span
                  class="c35"
                >
                  Create a new component
                </span>
              </button>
            </div>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="c41 c42"
    >
      <div />
    </div>
  </div>
  <div
    class="c43"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
